test/sunglasses-tests.js

```diff
@@ -0,0 +1,446 @@
+          done();
+        })
+    });
+
+    it('it should FAIL if there is an invalid Access Token Key.', done => {
+      chai
+        .request(server)
+        .get('/api/me/cart?accessToken=456421354')
+        .end((error, response) => {
+          response.should.have.status(403);
+          done();
+        })
+    });
+
+    it('it should GET the users cart successfully.', done => {
+      chai
+        .request(server)
+        .get(`/api/me/cart?accessToken=${currentAccessToken}`)
+        .end((error, response) => {
+          response.should.have.status(200);
+          response.body.should.be.an('array');
+          done();
+        })
+    });
+
+  });
+
+  describe ('POST/ api/me/cart', () => {
+
+    it('it should FAIL if there is no Access Token in the URL.', done => {
+      chai
+        .request(server)
+        .post('/api/me/cart')
+        .send(productToKeep)
+        .end((error, response) => {
+          response.should.have.status(401);
+          done();
+        })
+    });
+
+    it('it should FAIL if there is an invalid Access Token Key.', done => {
+      chai
+        .request(server)
+        .post('/api/me/cart?accessToken=456446541065421354')
+        .send(productToKeep)
+        .end((error, response) => {
+          response.should.have.status(403);
+          done();
+        })
+    });
+
+    it('it should FAIL if the product is missing from the request', done => {
+      chai
+        .request(server)
+        .post(`/api/me/cart?accessToken=${currentAccessToken}`)
+        .end((error, response) => {
+          response.should.have.status(415);
+          done();
+        })
+    });
+
+    it('it should FAIL if the product has invalid syntax.', done => {
+      chai
+        .request(server)
+        .post(`/api/me/cart?accessToken=${currentAccessToken}`)
+        .send(badlyFormattedProduct)
+        .end((error, response) => {
+          response.should.have.status(400);
+          done();
+        })
+    });
+
+    it('it should FAIL if the product sent is not in the database.', done => {
+      chai
+        .request(server)
+        .post(`/api/me/cart?accessToken=${currentAccessToken}`)
+        .send(missingProduct)
+        .end((error, response) => {
+          response.should.have.status(404);
+          done();
+        })
+    });
+
+    it('it should POST the product to the cart if the product sent is in the database (but not in the cart).', done => {
+      chai
+        .request(server)
+        .post(`/api/me/cart?accessToken=${currentAccessToken}`)
+        .send(productToKeep)
+        .end((error, response) => {
+          response.should.have.status(200);
+          response.body.should.be.an('array');
+          response.body.should.deep.include({'product':productToKeep, 'quantity':1});
+          done();
+        })
+    });
+
+    it('if the product is in the database AND is already in the cart, it should INCREASE the quantity by 1.', done => {
```

Comment:
This is an interesting way of writing this endpoint. It works, but I wonder if you might not also want to send the quantity with the request.